tests/framework/session/test_session_extension_hooks.py

```diff
@@ -526,6 +526,20 @@ def wait_and_identity(x: Any):
     return node(wait_and_identity, inputs="ds1", outputs="ds2", name="test-node")
 
 
+@pytest.fixture
+def sample_node_multiple_outputs():
+    def wait_and_identity(x: Any, y: Any):
+        time.sleep(0.1)
+        return (x, y)
```

Comment:
Why do we need the `wait_and_identity` function?

Reply:
Do we really need the `time.sleep()` part?

Reply:
I thought about this. The test will clearly work without this call, I thought to keep it because:

1. `_run_node_async` is designed to work specifically for nodes which block the thread, it makes sense to test it with such a node, even though it is not immediately clear what we gain from doing so.
2. It is a signal to people who want to modify the test / function in the future what the purpose of `_run_node_async` is.
3. The maximum gain from removing it is a decrease of 0.2 seconds in running our test suite.

I don't see a large advantage / disadvantage to either keeping or removing it, so I would be OK with either, though I don't think too much thought should be invested.